src/clients/BaseAbstractClient.ts

```diff
@@ -58,15 +59,21 @@ export abstract class BaseAbstractClient {
    * @provider Ethers RPC provider instance.
    * @returns An EventSearchConfig instance if valid, otherwise an UpdateFailureReason.
    */
-  public async updateSearchConfig(provider: providers.Provider): Promise<EventSearchConfig | UpdateFailureReason> {
+  public async updateSearchConfig(
+    provider: providers.Provider | Rpc<SolanaRpcApiFromTransport<RpcTransport>>
+  ): Promise<EventSearchConfig | UpdateFailureReason> {
     const fromBlock = this.firstBlockToSearch;
     let { toBlock } = this.eventSearchConfig;
     if (isDefined(toBlock)) {
       if (fromBlock > toBlock) {
         throw new Error(`Invalid event search config fromBlock (${fromBlock}) > toBlock (${toBlock})`);
       }
     } else {
-      toBlock = await provider.getBlockNumber();
+      if (provider instanceof providers.Provider) {
+        toBlock = await provider.getBlockNumber();
+      } else {
+        toBlock = Number(await provider.getBlockHeight({ commitment: "confirmed" }).send());
```

Comment:
@james-a-morris Did you check on the time delta for `confirmed` vs `processed`? I think the equivalent behaviour on EVM is `processed` since we typically don't want to impose any finality overheads on the data ingested here.

Reply:
This should be OK :+1: